yt_dlp/extractor/epicon.py

```diff
@@ -0,0 +1,119 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import ExtractorError
+
+
+class EpiconIE(InfoExtractor):
+    _VALID_URL = r'(?:https?://)(?:www\.)?epicon\.in/(?:documentaries|movies|tv-shows/[^/?#]+/[^/?#]+)/(?P<id>[^/?#]+)'
+    _TESTS = [{
+        'url': 'https://www.epicon.in/documentaries/air-battle-of-srinagar',
+        'info_dict': {
+            'id': 'air-battle-of-srinagar',
+            'ext': 'mp4',
+            'title': 'Air Battle of Srinagar',
+            'description': 'md5:c4de2013af9bc05ae4392e4115d518d7',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        }
+    }, {
+        'url': 'https://www.epicon.in/movies/krit',
+        'info_dict': {
+            'id': 'krit',
+            'ext': 'mp4',
+            'title': 'Krit',
+            'description': 'md5:c12b35dad915d48ccff7f013c79bab4a',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        }
+    }, {
+        'url': 'https://www.epicon.in/tv-shows/paapnaashini-ganga/season-1/vardaan',
+        'info_dict': {
+            'id': 'vardaan',
+            'ext': 'mp4',
+            'title': 'Ep 1 - VARDAAN',
+            'description': 'md5:f517058c3d0402398eefa6242f4dd6ae',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        }
+    }, {
+        'url': 'https://www.epicon.in/movies/jayadev',
+        'info_dict': {
+            'id': 'jayadev',
+            'ext': 'mp4',
+            'title': 'Jayadev',
+            'description': 'md5:09e349eecd8e585a3b6466904f19df6c',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        }
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        webpage = self._download_webpage(url, id)
+        cid = self._search_regex(r'class=\"mylist-icon\ iconclick\"\ id=\"(\d+)', webpage, 'cid')
+        headers = {'content-type': 'application/x-www-form-urlencoded; charset=UTF-8'}
+        data = f'cid={cid}&action=st&type=video'.encode()
+        data_json = self._parse_json(self._download_json('https://www.epicon.in/ajaxplayer/', id, headers=headers, data=data), id)
+
+        if not data_json['success']:
+            raise ExtractorError(data_json['message'], expected=True)
+
+        title = self._og_search_title(webpage).replace('Short Film ', '').replace('Watch ', '').replace(' |', '').replace(' Online at EPIC ON', '').replace(' Online On EPIC ON', '')
+        description = self._og_search_description(webpage) or None
+        thumbnail = self._og_search_thumbnail(webpage) or None
```

Comment:
why `or None`?

Reply:
The site is not very stable so i assumed it can be `''` and therefore added `or None` since these aren't fatal
Title and cid on the other hand are indeed fatal